src/cmd/api/main.go

```diff
@@ -87,6 +87,13 @@ func main() {
 	// Register service on consul (only for local env)
 	registerService(cfg, discoveryClient)
 
+	// Run watcher on services, it updates the transformation service cache if a new service
+	// is register/deregister on consul
+	go func() {
+		err := discoveryClient.Watch()
```

Comment:
Here I expect to have a context passed, so we can gracefully shutdown the watcher in case the program terminate.

Reply:
something like this https://github.com/sogilis/templates/blob/main/golang/concurrency/routines/context_sigint.go#L32